src/json-symtab-language/json_symbol_table.cpp

```diff
@@ -49,4 +49,6 @@ void symbol_table_from_json(const jsont &in, symbol_tablet &symbol_table)
         "symbol_table_from_json: duplicate symbol name `" +
         id2string(symbol.name) + "`");
   }
+
+  symbol_table.validate(validation_modet::EXCEPTION);
```

Comment:
So if the validation fails, this will throw a `incorrect_goto_program_exceptiont` - Is there an argument for saying this should be caught and re-thrown as a `deserialization_exceptiont` here? I honestly don't know, so I'm raising the question :-)

Reply:
It could go either way I think. We could argue that since we already have a complete symbol table that we validate it would be appropriate to throw an `incorrect_goto_program_exceptiont`, and that `deserialization_exceptiont` is for lower-level errors that are encountered during the construction (e.g., when we encounter an error while deserialising an `irep`).

Reply:
OK, fine by me.